src/Microsoft.ML.StandardLearners/Standard/SdcaStatic.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.ML.Data.StaticPipe.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Internal.Calibration;
+
+namespace Microsoft.ML.Runtime.Learners
+{
+    /// <summary>
+    /// Extension methods and utilities.
+    /// </summary>
+    public static class SdcaStatic
+    {
+        public static Scalar<float> PredictSdcaRegression(this Scalar<float> label, Vector<float> features, Scalar<float> weights = null,
+            float? l2Const = null,
+            float? l1Threshold = null,
+            float convergenceTolerance = 0.01f,
+            int? maxIterations = null,
+            bool shuffle = true,
+            float biasLearningRate = 1,
+            ISupportSdcaRegressionLossFactory loss = null,
+            Action<LinearRegressionPredictor> onFit = null)
+        {
+            var args = new SdcaRegressionTrainer.Arguments()
+            {
+                L2Const = l2Const,
+                L1Threshold = l1Threshold,
+                ConvergenceTolerance = convergenceTolerance,
+                MaxIterations = maxIterations,
+                Shuffle = shuffle,
+                BiasLearningRate = biasLearningRate,
+                LossFunction = loss ?? new SquaredLossFactory()
+            };
+
+            var rec = new TrainerEstimatorReconciler.Regression(
+                (env, labelName, featuresName, weightsName) =>
+                {
+                    var trainer = new SdcaRegressionTrainer(env, args, featuresName, labelName, weightsName);
+                    if (onFit != null)
+                        return trainer.WithOnFitDelegate(trans => onFit(trans.Model));
+                    return trainer;
+                }, label, features, weights);
+
+            return rec.Score;
+        }
+
+        public static (Scalar<float> score, Scalar<float> probability, Scalar<bool> predictedLabel)
+            PredictSdcaBinaryClassification(this Scalar<bool> label, Vector<float> features, Scalar<float> weights = null,
+                float? l2Const = null,
+                float? l1Threshold = null,
+                float convergenceTolerance = 0.1f,
+                int? maxIterations = null,
+                bool shuffle = true,
+                float biasLearningRate = 0,
+                Action<LinearBinaryPredictor, ParameterMixingCalibratedPredictor> onFit = null)
```

Comment:
>onFit [](start = 82, length = 5)

is this so we can pass different types of calibrators?

Reply:
No, this is so the user of this method can be informed about the calibrator, so they know the slope, etc.

---
In reply to: [215331242](https://github.com/dotnet/machinelearning/pull/837#discussion_r215331242) [](ancestors = 215331242)